ddtrace/internal/writer.py

```diff
@@ -225,20 +226,19 @@ def flush_queue(self):
         pass
 
 
-class AgentWriter(periodic.PeriodicService, TraceWriter):
-    """Writer to the Datadog Agent.
-
-    The Datadog Agent supports (at the time of writing this) receiving trace
-    payloads up to 50MB. A trace payload is just a list of traces and the agent
-    expects a trace to be complete. That is, all spans with the same trace_id
-    should be in the same trace.
-    """
+class HTTPWriter(periodic.PeriodicService, TraceWriter):
```

Comment:
This could be pulled out as a refactor that introduces a new abstract class or another interface that can be used for the ci agentless writer.

Reply:
done in #3690